event-stream-processing/src/parsers/application-classification.parser.ts

```diff
@@ -38,32 +40,45 @@ export class ApplicationClassificationParser implements Parser {
 
     for (const knownDomain of knownDomains) {
       if ((knownDomain.regex).test(urlDomain.toLowerCase())) {
-        lodash.set(document.data, 'service.name', knownDomain.app);
+        lodash.set(document.data, 'service.target.name', knownDomain.app);
       }
     }
 
-    if (lodash.isNil(lodash.get(document.data, 'service.name'))) {
-      const m = knownAppContextRegex.exec(urlPath);
-      if (m !== null && m.groups) {
-        for (const groupName of Object.keys(m.groups)) {
-          const fieldName = groupName.replace('__', '.');
-          const groupValue = m.groups[groupName].toLowerCase();
-          lodash.set(document.data, fieldName, groupValue);
+    if (lodash.isNil(lodash.get(document.data, 'service.target.name')) && urlPath.startsWith('/clp-cgi')) {
+      lodash.set(document.data, 'service.target.name', 'clp-cgi');
+    }
+
+    if (lodash.isNil(lodash.get(document.data, 'service.target.name'))) {
+      for (const regex of [knownAppContextRegex_v1, knownAppContextRegex_v2]) {
+        const m = regex.exec(urlPath);
+        if (m !== null && m.groups) {
+          for (const groupName of Object.keys(m.groups)) {
+            const fieldName = groupName.replace(/__/g, '.');
+            const groupValue = m.groups[groupName].toLowerCase();
+            lodash.set(document.data, fieldName, groupValue);
+          }
+          break;
+        }
+      }
+      if (lodash.isNil(lodash.get(document.data, 'service.target.name'))) {
+        if (!lodash.isNil(lodash.get(document.data, 'labels.project'))) {
+          const value = lodash.get(document.data, 'labels.project').toLowerCase();
+          lodash.set(document.data, 'service.target.name', value);
         }
       }
     }
-    if (lodash.isNil(lodash.get(document.data, 'service.name')) && urlPath.startsWith('/clp-cgi')) {
-      lodash.set(document.data, 'service.name', 'clp-cgi');
-    }
+
     // https://www.oracle-and-apex.com/apex-url-format/
-    if (lodash.get(document.data, 'service.name') === 'apex') {
+    /* eslint-disable max-len,camelcase,@typescript-eslint/no-unsafe-call */
+    if (lodash.get(document.data, 'service.target.name') === 'apex' || lodash.get(document.data, 'labels.project') === 'apex') {
```

Comment:
I think we'll need to add in a labels.target_project and labels.source_project to the index schema. It'll need to be in a separate PR so the index can rollover before we deploy anything that sets it.

Reply:
parsing labels.target_project and service.target.name from url.path